### PR TITLE
Hiero: enhancing effect publishing categorization - AY-5146

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
This adds a setting and code for using track names when categorizing the soft effects in NukeStudio/Hiero.

## Testing notes:
1. In NukeStudio add effects to a track item and move those into separate video tracks.
2. In `ayon+settings://hiero/publish/CollectClipEffects/effect_tracks` add a category and track regex that match 1 or more track names.
3. Setup publishing of the track item and the effects should be categorized from the settings.
4. Publish and validate the effects are correct in Nuke.

### CODE REVIEW

File: server_addon/hiero/server/settings/publish_plugins.py [hunk 1]
```diff
@@ -27,11 +27,21 @@ def validate_name(cls, value):
         return normalize_name(value)
 
 
+class CollectClipEffectsTracksModel(BaseSettingsModel):
+    _layout = "expanded"
+    name: str = SettingsField("", title="Name")
+    track_regex: str = SettingsField("", title="Track Regex")
```
Comment (jakubjezek001) [inline server_addon/hiero/server/settings/publish_plugins.py:33]: this should be different type > list[str] 

There will be cases where we will need to hove node classess from multiple tracks to be one product.

**Review by jakubjezek001 — APPROVED** [review]
Lets process suggested changes also please include screenshot of settings with example testing configuration. You were using for your testing.

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
Lets process suggested changes also please include screenshot of settings with example testing configuration. You were using for your testing.

File: server_addon/hiero/server/settings/publish_plugins.py (hunk 1 above)
Comment (tokejepsen) [reply server_addon/hiero/server/settings/publish_plugins.py:33]: We can target multiple tracks with a single regex. But if we are going down the route of multiple track names, then we might as well be explicit and not use regex, so something like `track_names` list where people can input `creative_look` and `invert_look` for example.

Comment (jakubjezek001) [reply server_addon/hiero/server/settings/publish_plugins.py:33]: yeah lets  just be explicit, then. Important is that there will be multiple track inputs.

Comment (tokejepsen) [reply server_addon/hiero/server/settings/publish_plugins.py]: Done

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
it does not work for me. 

here is my testing scene with two layers. 
image

You can see that it is creating 3 instances but it should really create only 2.
image

**Review by jakubjezek001 — APPROVED** [review]
works like a charm

**Pull request merged**